services/QuillLMS/app/controllers/teachers/classrooms_controller.rb

```diff
@@ -226,9 +235,11 @@ def classroom_params
   end
 
   def authorize_owner!
-    return unless params[:id].present?
```

Comment:
This line returns early if there is no params[:id]. Thus, this line triggers no redirect or render, meaning that the filter does not filter as expected.